teamcity/group.go

```diff
@@ -78,3 +95,13 @@ func (s *GroupService) Delete(key string) error {
 	err := s.restHelper.delete(locator, "group")
 	return err
 }
+
+// List - List of all groups
+func (s *GroupService) List() (*GroupList, error) {
```

Comment:
For the `List()` feature - At first this may seem an easy win to be able to return all items at once.
However, this project as whole has no support for paginated responses, and I think we should think more about implementing such support before returning potentially "unbounded" data sets.

Reasoning for this: In my experience, exposing such methods in a client is almost like an invitation for consumers/clients to abuse these calls.

I'd suggest to remove this unbounded `List()` unless there's a very strong case behind it to be supported, so we can resume the discussion about it.

For reference, see "Beware of partial answers..."
https://www.jetbrains.com/help/teamcity/rest-api.html#API+Client+Recommendations

Reply:
We can support splitting with `?locator=count:1,start: 3`

Reply:
```go
func (s *GroupService) List(start, count int) {
...
}
```

Reply:
I'd more inclined if we could provide this kind of support seamless for the client users, without needing to expose the pagination details. The response from such paginated methods would then be a sort of enumerator that could fetch all the results independently from the original method call.

This is something I thought about designing but haven't done the work yet.
I guess I'd be okay with the `List()` method without any pagination for this resource, at least until not we have a more general solution for the whole client.